masq/src/commands/set_exit_location_command.rs

```diff
@@ -0,0 +1,224 @@
+};
+use clap::{App, Arg, SubCommand};
+use masq_lib::as_any_ref_in_trait_impl;
+use masq_lib::messages::{CountryCodes, UiSetExitLocationRequest, UiSetExitLocationResponse};
+use masq_lib::shared_schema::common_validators;
+use masq_lib::utils::ExpectValue;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct SetExitLocationCommand {
+    pub exit_locations: Vec<CountryCodes>,
+    pub fallback_routing: bool,
+}
+
+impl SetExitLocationCommand {
+    pub fn new(pieces: &[String]) -> Result<Self, String> {
+        match set_exit_location_subcommand().get_matches_from_safe(pieces) {
+            Ok(matches) => {
+                let exit_locations = match matches.is_present("country-codes") {
+                    true => matches
+                        .value_of("country-codes")
+                        .expectv("required param")
+                        .split("|")
+                        .enumerate()
+                        .map(|(index, code)| CountryCodes::from((code.to_string(), index)))
+                        .collect(),
+                    false => vec![],
+                };
+                let fallback_routing = match (
+                    matches.is_present("fallback-routing"),
+                    matches.is_present("country-codes"),
+                ) {
+                    (false, true) => false,
+                    _ => true,
+                };
+                Ok(SetExitLocationCommand {
+                    exit_locations,
+                    fallback_routing,
+                })
+            }
+
+            Err(e) => Err(format!("SetExitLocationCommand {}", e)),
+        }
+    }
+}
+
+impl Command for SetExitLocationCommand {
+    fn execute(&self, context: &mut dyn CommandContext) -> Result<(), CommandError> {
+        let input = UiSetExitLocationRequest {
+            exit_locations: self.exit_locations.clone(),
+            fallback_routing: self.fallback_routing.clone(),
+        };
+
+        let _: UiSetExitLocationResponse =
+            transaction(input, context, STANDARD_COMMAND_TIMEOUT_MILLIS)?;
+        Ok(())
+    }
+
+    as_any_ref_in_trait_impl!();
+}
+
+const EXIT_LOCATION_ABOUT: &str =
+    "If you activate exit-location preferences, all exit Nodes in countries you don't specify will be prohibited: \n\
+    that is, if there is no exit Node available in any of your preferred countries, you'll get an error. However, \
+    if you just want to make a suggestion, and you don't mind Nodes in other countries being used if nothing is available \
+    in your preferred countries, you can specify --fallback-routing, and you'll get no error unless there are no exit Nodes \
+    available anywhere.\n\n\
+    Here are some example commands:\n\
+        masq> exit-location                                                       // disable exit-location \n\
+        masq> exit-location --fallback-routing                                    // disable exit-location \n\n\
```

Comment:
These comments sound like these commands remove the entire concept of exit locations. Maybe "disable exit-location preferences?"